homeassistant/components/mqtt/__init__.py

```diff
@@ -343,7 +343,7 @@ def async_setup(hass: HomeAssistantType, config: ConfigType):
             client_id = 'home-assistant'
     else:
         err = "Unable to start MQTT broker."
-        if conf.get(CONF_EMBEDDED) is not None:
+        if CONF_EMBEDDED in conf:
```

Comment:
This is not a style change. This is a lot different. Consider this `conf`:

```python
{
  CONF_EMBEDDED: None
}
```

Would not pass left, would pass right.

Reply:
And this is exactly why I am not a big fan of style changes.

Reply:
That's also why it's it's a "test" PR to gather feedback ;)

On the style change side I'm not completely of your opinion though. Sure, I don't have much experience with big code bases, but should any project just carry around old code and just do "fixes" when something breaks and/or implement features on top of the current code? Again, I'm no expert, but I think I've seen several projects fail because of code bloat.

And sure, occasionally (or more often unless there are such excellent reviewers 😉) something will break and users complain, but don't these things such as static typing (if you're referring to static typing) ultimately help *prevent* errors?

→ Just my humble opinion and I completely understand if this project wants to go this way, as there're also many positive aspects to this way.

Reply:
I have no problem with cleaning up code when we're doing functional changes in the same code. Also I'm not against adding typing (although it adds parsing and memory overhead), as it can help us find bugs + helps with IDE autocompletion (with again leads to fewer bugs).